mapping.json

```diff
@@ -1,232 +1,203 @@
             "mode": "field",
             "value": "itemID",
             "validation": {
                 "required": false
             }
-        },
-        {
+        }, {
             "name": "EndDate",
-            "mode": "field",
-            "value": "terminationDate",
+            "mode": "complex",
+            "value": "function addDays(date, days) {\r\n    let copy = new Date(Number(date));\r\n    copy.setDate(date.getDate() + days);\r\n    return copy;\r\n}\r\n\r\nfunction getEndDate() {\r\n    let endDate = null\r\n    let actualEndDate = null\r\n\r\n\r\n    if (typeof sourceContract.terminationDate !== 'undefined' && sourceContract.terminationDate) {\r\n        endDate = new Date(sourceContract.terminationDate);\r\n    }else{\r\n        endDate = new Date(\"2099-12-31\")\r\n    }\r\n\r\n    actualEndDate = addDays(endDate, 0);\r\n\r\n\r\n   return actualEndDate;\r\n\r\n}\r\n\r\ngetEndDate();",
             "validation": {
                 "required": false
             }
-        },
-        {
+        }, {
             "name": "ExternalId",
             "mode": "field",
             "value": "externalId",
             "validation": {
                 "required": true
             }
-        },
-        {
+        }, {
             "name": "Manager.ExternalId",
             "mode": "complex",
-            "value": "function getManagerExternalId(){\r\n    let managerExternalId = \"\";\r\n\r\n    if (typeof sourceContract.manager != 'undefined' && sourceContract.manager) {\r\n        if (typeof sourceContract.manager.workerID.idValue != 'undefined' && sourceContract.manager.workerID.idValue) {\r\n            managerExternalId = sourceContract.manager.workerID.idValue;\r\n        }\r\n    }\r\n\r\n    return managerExternalId;\r\n}\r\ngetManagerExternalId()",
+            "value": "function getManagerExternalId() {\r\n    let managerExternalId = \"\";\r\n    let managerItemID = \"\";\r\n\r\n    if (typeof sourceContract.manager != null && sourceContract.manager != \"\" && sourceContract.manager != undefined && sourceContract.manager) {\r\n\r\n        if ((sourceContract.manager.itemID).indexOf(\"HOOFD\") !== -1) {\r\n            managerExternalId = sourceContract.manager.workerID.idValue;\r\n        } else if ( typeof sourceContract.reportsTo != null && sourceContract.reportsTo != \"\" && sourceContract.reportsTo != undefined) {\r\n            if ((sourceContract.reportsTo).length >= 1) {\r\n                for (i = 0; i < (sourceContract.reportsTo).length; i++) {\r\n                    managerItemID = sourceContract.reportsTo[i].itemID;\r\n                    if ((managerItemID).indexOf(\"HOOFD\") !== -1) {\r\n                        managerExternalId = sourceContract.reportsTo[i].workerID.idValue;\r\n                        break;\r\n                    }\r\n                }\r\n\r\n            }\r\n        }\r\n    }\r\n\r\n\r\n    return managerExternalId;\r\n}\r\ngetManagerExternalId()",
```

Comment:
Is this something for the mapping or for in the powershell script? Maybe this should be moved to powershell?

Reply:
Correct!. Discussed with Mouki, the raw should be as pure as possible and calculation in the mapping